File: app/backend/src/schemas/admin_dashboard.py
from pydantic import BaseModel
from typing import List


class ActivityItemSchema(BaseModel):
    id: str
    message: str
    timeAgo: str


class WeeklyDataSchema(BaseModel):
    day: str
    count: int


class TopMetricsSchema(BaseModel):
    active_fires: int
    pending_approvals: int
    total_users: int
    system_status: str


class SystemMetricsSchema(BaseModel):
    predictions_completed: int
    model_health: str
    avg_confidence_percent: int
    last_sync_time: str


class DashboardSummaryResponse(BaseModel):
    top_metrics: TopMetricsSchema
    activity_log: List[ActivityItemSchema]
    weekly_incidents: List[WeeklyDataSchema]
    system_metrics: SystemMetricsSchema

File: app/frontend/src/types/AdminDashboard.ts
export interface ActivityItem {
    id: string;
    message: string;
    timeAgo: string;
}


export interface WeeklyData{
    day: string;
    count: number;
}

export interface TopMetrics{
    active_fires: number;
    pending_approvals: number;
    total_users: number;
    system_status: string;
}

export interface SystemMetrics{
    predictions_completed: number;
    model_health: string;
    avg_confidence_percent: number;
    last_sync_time: string;
}

export interface DashboardSummaryResponse{
    top_metrics: TopMetrics;
    activity_log: ActivityItem[];
    weekly_incidents: WeeklyData[];
    system_metrics: SystemMetrics;
}
